### PR TITLE
Lint Javascript & ERBs

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made?

When replacing the reports UI framework, I found the JS in Argo to be inconsistent and vague w/r/t expectations around style and formatting. We use linters in other languages and the amount of JS in Argo is substantial. Use the StandardJS linter rules.

Also, restore the erb_lint linter. I realized in setting up the JS linting that we apparently dropped ERB linting when we switched from TravisCI to CircleCI, which I do not believe was our intention. This updates erb_lint, adds it to rake & CI, and documents it in the README

# How was this change tested?

- [x] CI
- [x] Integration tests

### CODE REVIEW

**Review by edsu — DISMISSED** [review]
I think I understand why these aren't needed anymore, since eslint+standard replaces them?

* .hound.yml
* .jshintignore
* .jshintrc
* .scss-lint.yml
* babel.config.js

But do we not need postcss config anymore? Was it unused?

* postcss.config.js

**Review by edsu — COMMENTED** [review]
Switching back to a comment instead of approve since I just noticed it's on hold.

File: .browserslistrc [hunk 1]
```diff
@@ -1 +0,0 @@
-defaults
```
Comment (mjgiarlo) [inline .browserslistrc:1]: This file was added when we webpacker-ified Argo, and we longer use webpacker for assets.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
+  env: {
+    browser: true,
+    es2021: true
+  },
+  extends: 'standard',
```
Comment (mjgiarlo) [inline .eslintrc.js:6]: Use `eslint` tool with standard style guide in favor of maintaining custom lint configs.

File: .hound.yml [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-
-erblint:
-  enabled: true
-
-jshint:
-  config_file: .jshintrc
-  ignore_file: .jshintignore
-
-scss:
-  config_file: .scss-lint.yml
```
Comment (mjgiarlo) [inline .hound.yml:15]: We no longer use HoundCI.

File: .jshintignore [hunk 4]
```diff
@@ -1,3 +0,0 @@
-vendor/**
-app/assets/javascripts/vendor/**
-app/assets/javascripts/stimulus.js
```
Comment (mjgiarlo) [inline .jshintignore:3]: This was used by HoundCI, which we no longer use.

File: .jshintrc [hunk 5]
```diff
@@ -1,21 +0,0 @@
...
-
-  "latedef": true,
-
-  "strict": true,
-  "undef": true,
-  "unused": true,
-
-  "eqnull": true,
-  "predef": [
-    "expect", "describe", "it", "xit", "spyOn", "fixture", "beforeEach", "afterEach", "L", "jasmine", "TestResponses"
```
Comment (mjgiarlo) [inline .jshintrc:19]: This was used by HoundCI which we no longer use

File: .scss-lint.yml [hunk 6]
```diff
@@ -1,145 +0,0 @@
-scss_files: "**/*.scss"
```
Comment (mjgiarlo) [inline .scss-lint.yml:1]: This was only used by HoundCI, which Argo doesn't use any longer.

File: app/javascript/controllers/access_rights_controller.js [hunk 7]
```diff
@@ -2,23 +2,31 @@ import { Controller } from '@hotwired/stimulus'
 
 // This handles editing the access rights for a DRO
 export default class extends Controller {
-  static targets = ["view", "download", "location", "cdl", "downloadRow", "locationRow", "cdlRow"]
```
Comment (mjgiarlo) [inline app/javascript/controllers/access_rights_controller.js:5]: Changes to `app/javascript/**/*.js` are to bring the code in line with the linter.

File: babel.config.js [hunk 8]
```diff
@@ -1,72 +0,0 @@
-module.exports = function(api) {
-  var validEnv = ['development', 'test', 'production']
```
Comment (mjgiarlo) [inline babel.config.js:2]: This file was added when we webpacker-ified Argo, and we longer use webpacker for assets.

File: postcss.config.js [hunk 9]
```diff
@@ -1,12 +0,0 @@
-module.exports = {
```
Comment (mjgiarlo) [inline postcss.config.js:1]: This file was added when we webpacker-ified Argo, and we longer use webpacker for assets.

File: .gitattributes [hunk 10]
```diff
@@ -0,0 +1,3 @@
+Gemfile.lock linguist-generated=true
+yarn.lock linguist-generated=true
+.rubocop_todo.yml linguist-generated=true
```
Comment (mjgiarlo) [inline .gitattributes:3]: https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github

We don't need to be reviewing generated files.

File: .circleci/config.yml [hunk 11]
```diff
@@ -193,6 +195,24 @@ jobs:
...
       - ruby/rubocop-check
+  erblint:
+    executor: default
+    steps:
+      - checkout
+      - ruby/install-deps:
+          key: gems-v2
+      - run:
+          name: Lint ERBs
+          command: bundle exec erblint --lint-all --format compact
```
Comment (mjgiarlo) [inline .circleci/config.yml:206]: While adding the JS linting, I discovered that we have an ERB linter we no longer use. Why? It looks like we missed it when switching from TravisCI to CircleCI, which I believe was probably not intended.

File: .erb-lint.yml [hunk 12]
```diff
@@ -1,61 +1,36 @@
 ---
 exclude:
   - "**/vendor/bundle/**"
-EnableDefaultLinters: false
 linters:
-  ErbSafety:
```
Comment (mjgiarlo) [inline .erb-lint.yml:6]: Update to latest erblint config.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**